src/main/java/org/owasp/benchmark/helpers/Utils.java

```diff
@@ -0,0 +1,590 @@
+			boolean append) {
+		boolean result = true;
+		PrintStream os = null;
+		try {
+			Files.createDirectories(pathToFileDir);
+			File f = new File(completeName);
+			if (!append) {
+				f.delete();
+			}
+			FileOutputStream fos = new FileOutputStream(f, append);
+			os = new PrintStream(fos);
+
+			for (String ol : outputLines) {
+				os.println(ol);
+			}
+
+		} catch (IOException e1) {
+			result = false;
+			e1.printStackTrace();
+		} finally {
+			os.close();
+		}
+
+		return result;
+	}
+
+	public static boolean writeLineToFile(Path pathToFileDir, String completeName, String line) {
+		boolean result = true;
+		PrintStream os = null;
+		try {
+			Files.createDirectories(pathToFileDir);
+			File f = new File(completeName);
+			if (!f.exists()) {
+				// System.out.println("Archivo llamado: " +
+				// f.getAbsolutePath());
+				f.createNewFile();
+				// System.out.println("Archivo no existia se creoooooooooo");
+			}
+			FileOutputStream fos = new FileOutputStream(f, true);
+			os = new PrintStream(fos);
+			os.println(line);
+		} catch (IOException e1) {
+			result = false;
+			e1.printStackTrace();
+		} finally {
+			os.close();
+		}
+
+		return result;
+	}
+
+	public static boolean deleteFile(String completeName) {
+		boolean result = true;
+		File f = new File(completeName);
+		if (f.exists()) {
+			try {
+				f.delete();
+			} catch (SecurityException e) {
+				System.out.println("Can't delete file: " + completeName);
+				result = false;
+			}
+		}
+		return result;
+	}
+
+	public static List<AbstractTestCaseRequest> parseHttpFile(InputStream http, List<String> failedTestCases)
+			throws Exception {
+		String URL = "";
+
+		DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder docBuilder = docBuilderFactory.newDocumentBuilder();
+		InputSource is = new InputSource(http);
+		Document doc = docBuilder.parse(is);
+		Node root = doc.getDocumentElement();
+
+		DocumentBuilderFactory newCrawlerBF = null;
+		DocumentBuilder newCrawlerBuilder = null;
+		Document newCrawlerDoc = null;
+		Element newCrawkerRootElement = null;
+		newCrawlerBF = DocumentBuilderFactory.newInstance();
+		Node newNode;
+
+		try {
+			newCrawlerBuilder = newCrawlerBF.newDocumentBuilder();
+		} catch (ParserConfigurationException e) {
+			System.out.println("Problem init the Crawler XML file");
+		}
+		newCrawlerDoc = newCrawlerBuilder.newDocument();
+		newCrawkerRootElement = newCrawlerDoc.createElement("benchmarkSuite");
+		newCrawlerDoc.appendChild(newCrawkerRootElement);
+
+		List<AbstractTestCaseRequest> requests = new ArrayList<AbstractTestCaseRequest>();
+		List<Node> tests = XMLCrawler.getNamedChildren("benchmarkTest", root);
+		for (Node test : tests) {
+			URL = XMLCrawler.getAttributeValue("URL", test).trim();
+			if (failedTestCases
+					.contains(URL.substring(URL.indexOf("BenchmarkTest"), URL.indexOf("BenchmarkTest") + 18))) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 448 does not find the substring and returns -1, the call to substring() on line 448 will throw IndexOutOfBoundsException.